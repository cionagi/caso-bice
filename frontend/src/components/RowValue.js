import React from 'react';
import PropTypes from 'prop-types'; // ES6

const RowValue = ({ valueData }) => {
  const { key, name, unit, date, value } = valueData;
  return (
    <div className="row">
      <div className="col-12">
        <div className="border my-2 py-3 px-2 d-flex justify-content-between align-items-baseline bg-light">
          <div className="text-body">{key}</div>
          <div className="text-body">{name}</div>
          <div className="text-body">{unit}</div>
          <div className="text-body">{date} </div>
          <div className="text-body">{value} </div>
          <div>
            <button type="button" className="btn btn-primary">
              Ver detalles
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

RowValue.propTypes = {
  valueData: PropTypes.object,
};

export default RowValue;
