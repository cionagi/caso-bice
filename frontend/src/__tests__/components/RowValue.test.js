import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import RowValue from '../../components/RowValue';

const FAKE_DATA = {
  key: 1,
  name: 'Cobre',
  dateParse: '11-04-2020',
  valueParse: '1.000',
};

test('should render correctly', () => {
  const { asFragment } = render(<RowValue valueData={FAKE_DATA} />);
  expect(asFragment()).toMatchSnapshot();
});

test('should render correctly', () => {
  const { getByTestId } = render(<RowValue valueData={FAKE_DATA} />);
  const rightClick = { button: 2 };
  fireEvent.click(getByTestId('btn-get-historic'), rightClick);
});
