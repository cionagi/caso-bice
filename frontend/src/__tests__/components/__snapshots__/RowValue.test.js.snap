// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="border my-2 py-3 px-2 bg-light text-body"
      >
        <div
          class="row justify-content-between text-center"
        >
          <div
            class="col-12 col-md-3 d-flex align-items-center font-weight-bold"
          >
            Cobre
          </div>
          <div
            class="col-12 col-md-3  d-flex align-items-center"
          >
            1.000
          </div>
          <div
            class="col-12 col-md-3 col-md-3 d-flex align-items-center"
          >
            11-04-2020
          </div>
          <div
            class="col-12 col-md-3 d-flex align-items-center"
          >
            <button
              class="btn  btn-custom"
              type="button"
            >
              Historico
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-12"
          >
            <div
              class=" mt-3 collapse undefined"
            >
              <div
                class="card card-body"
              >
                <div
                  class="d-flex justify-content-center align-items-center"
                  data-testid="loading-component"
                >
                  <div
                    class="spinner-border"
                    role="status"
                    style="width: 5rem; height: 5rem;"
                  >
                    <span
                      class="sr-only"
                    >
                      Loading...
                    </span>
                  </div>
                </div>
                <div
                  class="row justify-content-center"
                >
                  <div
                    class="col-12 col-md-8 mt-3 text-center d-flex justify-content-center"
                  >
                    <nav>
                      <ul
                        class="pagination pagination-sm"
                        data-testid="pagination"
                      />
                    </nav>
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-12 col-md-12"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
